edweiss-app/app/(app)/courses/[id]/index.tsx

```diff
@@ -83,31 +85,24 @@ const CoursePage: ApplicationRoute = () => {
 	const isValidId = typeof id === 'string';
 
 	// Récupérer les données du cours et des assignments depuis Firestore
-	const [course] = usePrefetchedDynamicDoc(
+	const result = usePrefetchedDynamicDoc(
 		CollectionOf<Course>('courses'),
 		isValidId ? id : '',
 		undefined
 	);
 
+	const [course] = Array.isArray(result) ? result : [undefined];
+
 	const firebase_data = useDynamicDocs(
 		CollectionOf<Assignment>(`courses/${isValidId ? id : ''}/assignments`)
-	);
+	) || [];
 
-	if (!isValidId) {
-		return <Redirect href={'/'} />;
-	}
+	if (!isValidId) { return <Redirect href={'/'} />; }
 
 	// Afficher un indicateur de chargement si les données ne sont pas encore prêtes
-	if (course == undefined || firebase_data == undefined) {
-		return <TActivityIndicator size={40} />;
-	}
+	if (course == undefined || firebase_data == undefined) { return <TActivityIndicator size={40} />; }
 
-	// Get course & assignments data from Firestore
-	//const [course] = usePrefetchedDynamicDoc(CollectionOf<Course>('courses'), id, undefined);
-	//const firebase_data = useDynamicDocs(CollectionOf<Assignment>(`courses/${id}/assignments`));
-	//if (course == undefined || firebase_data == undefined) return <TActivityIndicator size={40} />;
-
-	const assignmentsArray = firebase_data ? firebase_data.map(doc => ({ id: doc.id, data: doc.data })) : [];
+	const assignmentsArray = firebase_data.map(doc => ({ id: doc.id, data: doc.data }));
```

Comment:
Why are you doing this? `firebase_data` is already a Document<Assignment>[], you are reassigning the same fields to themselves here.

Reply:
Yes you are completely right, actually in my mind This line was "unpacking" the array but it is not useful in fact. Thanks for pointing out that!!!!